### PR TITLE
Merge benchmark scripts into the main branch

### PR DESCRIPTION
**Pull request by frgossen**
Today, people often use slightly different benchmark setups, which leads to results that are hard to compare and hard to reproduce. Merge these benchmark scripts into the main branch so that we have a common ground for benchmarking.

### CODE REVIEW

**Comment by zpcore** [comment]
LGTM, thanks @frgossen  for driving the effort! Also added @miladm here since we want to merge it into the main branch.

 By the way, can we check in (https://github.com/pytorch/xla/pull/5907) before the merge?

**Comment by zpcore** [comment]
Hi @frgossen , I tried to run the script, but it gives the following error:
```
(pytorch) root@5f809d41da8f:~/xla/benchmarks# python experiment_runner.py --suite-name=torchbench --accelerator=tpu --xla=PJRT --progress-bar                      
Traceback (most recent call last):
  File "experiment_runner.py", line 18, in <module>
    from .benchmark_model import ModelLoader
ImportError: attempted relative import with no known parent package

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "experiment_runner.py", line 28, in <module>
    from tiers import append_filter_by_tier
  File "/root/xla_bm/benchmarks/tiers.py", line 11, in <module>
    def append_filter_by_tier(filter_list: list[str], filter_by_tier: list[int]):
TypeError: 'type' object is not subscriptable
```
Previously this will allow us to run all the installed models by default, can you fix that? And if we want to run a single model for example, we can use `--filter=BERT_pytorch`.

**Comment by JackCaoG** [comment]
This is great, can we have a short README on how to use these scripts?

File: benchmarks/run_benchmark.sh
Comment (ManfeiBai) [inline benchmarks/run_benchmark.sh:67]: deleted the command to specific device

**Review by ManfeiBai — APPROVED** [review]
LGTM

File: .vscode/settings.json
Comment (vanbasten23) [inline .vscode/settings.json:21]: I don't think this change is necessary.

Can you add this dir `benchmarks` to https://github.com/pytorch/xla/blob/175e86b82bc0e89a875903e424385ae2adce1bc7/.github/workflows/lintercheck.yml#L69 so CI change check the linter? We usually run `yapf -i -r *.py test/ scripts/ torch_xla/ benchmarks/` to fix the linter

**Comment by vanbasten23** [comment]
Thanks @frgossen for doing it. Overall it LGTM with a minor comment. Also please check @zpcore and @JackCaoG 's comment above.

File: .vscode/settings.json
Comment (frgossen) [reply .vscode/settings.json:21]: Thanks!

**Comment by frgossen** [comment]
> This is great, can we have a short README on how to use these scripts?

Done.

**Comment by frgossen** [comment]
> Hi @frgossen , I tried to run the script, but it gives the following error:
> 
> ```
> (pytorch) root@5f809d41da8f:~/xla/benchmarks# python experiment_runner.py --suite-name=torchbench --accelerator=tpu --xla=PJRT --progress-bar                      
> Traceback (most recent call last):
>   File "experiment_runner.py", line 18, in <module>
>     from .benchmark_model import ModelLoader
> ImportError: attempted relative import with no known parent package
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "experiment_runner.py", line 28, in <module>
>     from tiers import append_filter_by_tier
>   File "/root/xla_bm/benchmarks/tiers.py", line 11, in <module>
>     def append_filter_by_tier(filter_list: list[str], filter_by_tier: list[int]):
> TypeError: 'type' object is not subscriptable
> ```
> 
> Previously this will allow us to run all the installed models by default, can you fix that? And if we want to run a single model for example, we can use `--filter=BERT_pytorch`.

`--filter=BERT_pytorch` should work already. You might have to make it `--filter=^BERT_pytorch$` if you don't want substring matches. 
As for the error, it should all work if you run it from the pytorch directory. I added that to the readme.

**Comment by zpcore** [comment]
> > Hi @frgossen , I tried to run the script, but it gives the following error:
> > ```
> > (pytorch) root@5f809d41da8f:~/xla/benchmarks# python experiment_runner.py --suite-name=torchbench --accelerator=tpu --xla=PJRT --progress-bar                      
> > Traceback (most recent call last):
> >   File "experiment_runner.py", line 18, in <module>
> >     from .benchmark_model import ModelLoader
> > ImportError: attempted relative import with no known parent package
> > 
> > During handling of the above exception, another exception occurred:
> > 
> > Traceback (most recent call last):
> >   File "experiment_runner.py", line 28, in <module>
> >     from tiers import append_filter_by_tier
> >   File "/root/xla_bm/benchmarks/tiers.py", line 11, in <module>
> >     def append_filter_by_tier(filter_list: list[str], filter_by_tier: list[int]):
> > TypeError: 'type' object is not subscriptable
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Previously this will allow us to run all the installed models by default, can you fix that? And if we want to run a single model for example, we can use `--filter=BERT_pytorch`.
> 
> `--filter=BERT_pytorch` should work already. You might have to make it `--filter=^BERT_pytorch$` if you don't want substring matches. As for the error, it should all work if you run it from the pytorch directory. I added that to the readme.

Thanks, I synced the codebase to branch `benchmark: 56e57f0fe5a6bd9300464415a7e3172a2c7dd57f`, and run outside the xla folder:
```
$ python xla/benchmarks/experiment_runner.py --suite-name=torchbench --accelerator=tpu --xla=PJRT --progress-bar --filter=^BERT_pytorch$
```
Still it reports the same issue. Not sure if others see the issue or not.

**Comment by zpcore** [comment]
Hi @frgossen , can you try with torch_xla 2.2 build? In 2.2 build, the script will report `ClusterResolver` is not defined, 2.0 release doesn't show the issue though. https://github.com/pytorch/xla/pull/5907 should remove the dependency of `ClusterResolver` also.

**Comment by frgossen** [comment]
> Thanks, I synced the codebase to branch benchmark: 56e57f0fe5a6bd9300464415a7e3172a2c7dd57f, and run outside the xla folder:
This is strange. I ran the exact command and for me it only fails because for a lack of TPU. 
`python xla/benchmarks/experiment_runner.py --suite-name=torchbench --accelerator=cuda  --xla=PJRT --progress-bar` starts benchmarking as expected.

**Comment by frgossen** [comment]
> Hi @frgossen , can you try with torch_xla 2.2 build? In 2.2 build, the script will report `ClusterResolver` is not defined, 2.0 release doesn't show the issue though. #5907 should remove the dependency of `ClusterResolver` also.
> 
> @RissyRan may also see this issue before since she is the author updated `ClusterResolver` dependency.

Added your `ClusterResolver` patch. This is TPU-specific, so, unfortunately, I cannot test this atm.

**Comment by frgossen** [comment]
Would you mind merging https://github.com/pytorch/xla/pull/5907 onto the main branch once this has landed. I see there are still comments on the PR.

**Comment by zpcore** [comment]
> Would you mind merging #5907 onto the main branch once this has landed. I see there are still comments on the PR.

I agree, let's merge the benchmark branch to master first since those scripts will not break anything at this time. We can check any potential issue more easily once the branch is merged. Thanks for the effort!

**Comment by miladm** [comment]
would be great to merge this PR today if at all possible. thank you @zpcore

**Pull request merged**
